java/src/test/java/org/rocksdb/RocksDBTest.java

```diff
@@ -602,6 +602,60 @@ public void compactRangeToLevel()
     }
   }
 
+  @Test
+  public void deleteFilesInRange()
+      throws RocksDBException, InterruptedException {
+    final int KEY_SIZE = 20;
+    final int VALUE_SIZE = 3000;
+    final int FILE_SIZE = 100000;
+    final int NUM_FILES = 10;
+    
+    final int KEY_INTERVAL = 10000;
+    try (final Options opt = new Options().
+        setCreateIfMissing(true).
+        setCompactionStyle(CompactionStyle.LEVEL).
+        setCompressionType(CompressionType.NO_COMPRESSION).
+        setTargetFileSizeBase(FILE_SIZE).
+        setTargetFileSizeMultiplier(1).
+        setWriteBufferSize(FILE_SIZE / 2).
+        setNumLevels(5).
+        setDisableAutoCompactions(true);
+         final RocksDB db = RocksDB.open(opt,
+             dbFolder.getRoot().getAbsolutePath())
+    ) {
+      int records = FILE_SIZE / (KEY_SIZE + VALUE_SIZE);
+
+      // fill database with key/value pairs
+      byte[] value = new byte[VALUE_SIZE];
+      int key_init = 0;
+      for (int o = 0; o < NUM_FILES; ++o) {
+        int int_key = key_init++;
+        for (int i = 0; i < records; ++i) {
+          int_key += KEY_INTERVAL;
+          rand.nextBytes(value);
+  
+          db.put(String.format("%020d", int_key).getBytes(),
+              value);
+        }
+      }
+      db.flush(new FlushOptions().setWaitForFlush(true));
+      db.compactRange();
+      // Make sure we do create one more L0 files.
+      assertThat(
+          db.getProperty("rocksdb.num-files-at-level0")).
+          isEqualTo("0");
+
+      // Should be 10
+      int files = Integer.parseInt(db.getProperty("rocksdb.num-files-at-level1"));
+      assertThat(files).isBetween(8, 12);
```

Comment:
The comment states this should be `10` but you are asserting `8 to 12`?

Reply:
Yes. I am not asserting exact numbers, but that delete files actually deletes something. I did not want to introduce unstable test.

Reply:
So, would the instability would come due to the "compactions are not predicable" issue you mentioned?

Reply:
Yes. Sometimes even if you specify targetSize it is using memory log on level 0 and ignoring target size. Actually compaction then just promotes file to higher level, because there is no overlap thus size is bigger than expected for test. That is why I am doing that crazy interleaved puts to force worst scenario so that compaction end up with "nice" files I can delete. Because we are not trying to assert logic in C++ I am fine with these acceptable ranges.